dataplane/Dockerfile

```diff
@@ -7,14 +7,20 @@ RUN rustup default stable
 RUN rustup install nightly
 RUN rustup component add rust-src --toolchain nightly
 RUN rustup target add x86_64-unknown-linux-musl
-RUN cargo install bpf-linker
+RUN --mount=type=cache,target=/root/.cargo/registry \
+    cargo install bpf-linker
 
 WORKDIR /workspace
 
 COPY . .
-RUN --mount=type=cache,target=/workspace/target/ cargo xtask build-ebpf --release
-RUN --mount=type=cache,target=/workspace/target/ RUSTFLAGS=-Ctarget-feature=+crt-static cargo build --release --target=x86_64-unknown-linux-musl
-RUN --mount=type=cache,target=/workspace/target/ cp /workspace/target/x86_64-unknown-linux-musl/release/loader /workspace/dataplane
+RUN --mount=type=cache,target=/workspace/target/ \
+    --mount=type=cache,target=/root/.cargo/registry \
+    cargo xtask build-ebpf --release
+RUN --mount=type=cache,target=/workspace/target/ \
+    --mount=type=cache,target=/root/.cargo/registry \
+    RUSTFLAGS=-Ctarget-feature=+crt-static cargo build --release --target=x86_64-unknown-linux-musl
+RUN --mount=type=cache,target=/workspace/target/ \
```

Comment:
Could you just copy the binary on line 32 from the builder stage directly rather than doing the extra copy here?

Reply:
Unfortunately no because of the use of the buildkit cache. The final binary is actually not present in the filesystem layers of the image, UNLESS I copy it out of the build cache. I could probably do some other tricks to make this work, like caching differently but I think this is the simplest solution.